Added GPSEstimation container and handler for the platform.

Draskown opened the pull request:
So I finished the rewrtiting of my code in Java.

GPS coordinates should be added to the arrayOfTrueGPS in order to be handled.

Draskown committed:
Added GPSEstimation handlers for the platform.

XxOinvizioNxX commented:
Please implement the changes corresponding to the new release

XxOinvizioNxX commented:
And also please add your name to the copyright header in your files

Draskown committed:
Update README.md

F33RNI committed:
Merge branch 'main' into Draskown-commit-1

Draskown committed:
Implementation of a GPSEstimation algorithm

What I did is made two GPSEstimation classes: container and handler. 
Container contains 2 arrays of GPS coordinates: received and esitmated; and multiplier of estimation scale.

Draskown committed:
Mustache man

Did the tweaks for compilation to work
Hopefully..

Draskown committed:
Wrong file

Long story short - all this time I have been editing the wrong file.

Draskown committed:
Travis fixes

Draskown committed:
Travis fixes

XxOinvizioNxX commented:
![image](https://user-images.githubusercontent.com/39090123/118694627-eec9ee80-b814-11eb-9a7a-10a58b1cbe9e.png)

Draskown committed:
Crutches me dudhe

Draskown committed:
Crutches me dudhe

Draskown committed:
Crutches me dudhe

Draskown committed:
The right commit

Draskown committed:
The right commit

Draskown committed:
The right commit

Draskown committed:
The right commit

XxOinvizioNxX commented:
![image](https://user-images.githubusercontent.com/39090123/118696522-0bffbc80-b817-11eb-8d80-3a979e26364b.png)

Draskown committed:
The right commit

Draskown commented:
bruh

Draskown committed:
The right commit

F33RNI reviewed (commented):
KALachov-style edits 💩

1. Вместо жёсткого порога `acceptableDistance` сделать алгоритм с подобием филтра Калмана:
` координаты_на_дрон = предсказанные_координаты * K + координаты_платформы * (K-1)`
`K = 0` когда расстояние между дроном и платформой 0. `K = 1` когда расстояние больше некого порога (в настройках). Порог тут чисто для масштаба. Для преобразования диапазонов советую функцию map:
https://github.com/XxOinvizioNxX/Liberty-Way/blob/main/src/com/liberty_amls/WebAPI.java#L510-L516
Только переписать её под нужный тип данных

2. `earthRadiusM` заменить на `planetRadius` и внести в настройки (`SettingsContainer`) (включая `settings.json`). В `SettingsHandler` не забыть сделать парсинг этого значения (как и порога (маштаба) для пункта 1)

3. Убрать `GPSEstimationContainer`. Вместо него все 3 переменные хранить в handler. получение / установку данных сделать public методами в самом `GPSEstimationHandler`

4. Никакого `Velocity` класса! Скорость хранить сразу в _км/ч_ или _м/с_. в типе `double` внутри класса `TelemetryContainer`

5. Скорость считать используя лишь предыдущие координаты и время. Время советую использовать `long telemetryLastPacketTime`
https://github.com/XxOinvizioNxX/Liberty-Way/blob/main/src/com/liberty_amls/TelemetryHandler.java#L38
т.к. обновляется оно как раз в конце после успешного получения пакета
https://github.com/XxOinvizioNxX/Liberty-Way/blob/main/src/com/liberty_amls/TelemetryHandler.java#L188

6. Убрать `gpsEstimationContainer` из `TelemetryHandler`. (Заменить на какой-нибудь public void setGPS...(); внутри  `GPSEstimationHandler`) и использовать уже его в  `gpsEstimationContainer`

7. В `WebAPI` переменную `gpsEstimationHandler` сделать локальной (без объявления сверху)

8. По возможности, добавить вычисление скорости платформы в `PlatformHandler` (добавить переменную speedGPS в `PlatformCaontainer`.) Текущее получение скорости с платформы пока не трогать

Draskown committed:
Boop

F33RNI committed:
Move GPSEstimationHandler initialization

Move GPSEstimationHandler initialization directly to PositionHandler class initialization (line 316)

F33RNI committed:
Delete GPSEstimationContainer.java

Draskown committed:
Added proper distance calculation

Draskown committed:
Some tweaks

Mostly, regarding local variables and loop time calculating

F33RNI committed:
Ahh shit here we go again...

TelemetryHandler and PositionHandler initialization updates

F33RNI committed:
Some optimization

F33RNI reviewed (commented):
Please use arrays instead of lastGPSLat and lastGPSLon

F33RNI committed:
Fixed settingsContainer.planetRadius checking

F33RNI committed:
Changed speed initilization

And also check travis-ci

F33RNI committed:
Update .travis.yml

F33RNI committed:
Update .travis.yml

F33RNI committed:
Update .travis.yml

F33RNI committed:
Update .travis.yml

F33RNI committed:
Update .travis.yml

XxOinvizioNxX commented:
Now Travis-ci only builds the main branch

F33RNI committed:
Delete .travis.yml

F33RNI committed:
Create .travis.yml

F33RNI committed:
Update .travis.yml

F33RNI committed:
Revert "Update .travis.yml"

This reverts commit d51e4d22ac157edd9e541428479f8152a1a556e6.

F33RNI committed:
Revert "Create .travis.yml"

This reverts commit c61bebf70448d8f98f774f959ac2c41f00c9c3f6.

F33RNI committed:
Revert "Delete .travis.yml"

This reverts commit 705814f1c4112ebb9ca411919ca9044fd668f495.

F33RNI committed:
Revert "Update .travis.yml"

This reverts commit c4c9af3cd7f796d88e5adca0bbc05a4c4e122250.

F33RNI committed:
Revert "Update .travis.yml"

This reverts commit cf548f41e95d1d56ad2a95e2b97d552c680015f0.

F33RNI committed:
Revert "Update .travis.yml"

This reverts commit 41c623bf292c7f2c434113ddc1ff97ea3652c6a2.

F33RNI committed:
Revert "Update .travis.yml"

This reverts commit f64427d1590111f55422b8b92258e355eb15b108.

F33RNI committed:
Revert "Update .travis.yml"

This reverts commit 5c9eaad1d115e9005ba61a8fc81a0a8723be5767.

F33RNI committed:
Update .travis.yml

XxOinvizioNxX commented:
Now Travis-ci only builds the main branch

F33RNI committed:
Merge pull request #6 from XxOinvizioNxX/main

Update README.md and .travis.yml

XxOinvizioNxX commented:
Now Travis-ci only builds the main branch

Draskown committed:
Array usage

LastLat and LastLon are now taken from the trueGPS array

F33RNI commented on src/com/liberty_amls/SettingsHandler.java:
```diff
@@ -223,6 +223,17 @@ public void parseSettings() {
+            settingsContainer.notAcceptableDistance = jsonSettings.get("not_acceptable_distance").getAsInt();
+
+            // Debug option for whether to allow prediction of GPS coordinates or not
+            settingsContainer.allowPrediction = jsonSettings.get("allow_prediction").getAsBoolean();
```
Please rename `allowPrediction` to `GPSPredictionAllowed` and `gps_prediction_allowed`. Because it is not clear what kind of prediction we are talking about here

F33RNI replied:
Also, the style must match `landing_allowed`

F33RNI reviewed (commented):
Please remove fillArrays method, because PlatformHandler also has PlatformContainer. You can directly append your arrays

F33RNI commented on src/com/liberty_amls/PlatformHandler.java:
```diff
@@ -335,4 +340,42 @@ public void stop() {
+    private void calculateSpeed(){
+        double loopTime = System.currentTimeMillis() - platformLastPacketTime;
+
+        double distance = Math.sin((platformContainer.trueGPSLat.get(platformContainer.trueGPSLat.size() - 1) -
```
When I talked about reducing the number of local variables, I meant only those places where they are not needed. It is too irrational to count the size of the array every time. Please declare a variable for this

Draskown committed:
Added some variables and small changes

I fixed GPSPredictionAllowed setting, removed FillArays method

Draskown committed:
loopTime fix

From double to long

Draskown committed:
Moved calculation into the if statement

XxOinvizioNxX commented:
Please add `planet_radius`, `not_acceptable_distance` and `gps_prediction_allowed` to `settings.json` file in your branch

Draskown committed:
update settings.json

XxOinvizioNxX commented:
So, is that all? Can I finally merge these after three hundred edits? 

![image](https://user-images.githubusercontent.com/39090123/120028896-0451cc00-bffe-11eb-8ffb-6d16cd7f3f47.png)

Draskown commented:
Yeah, I hope so.
